dist/index.mjs

```diff
@@ -9,81 +9,85 @@ const defaultOptions = [
     id: 0,
     value: "Too weak",
     minDiversity: 0,
-    minLength: 0
+    minLength: 0,
   },
   {
     id: 1,
     value: "Weak",
     minDiversity: 2,
-    minLength: 6
+    minLength: 8,
   },
   {
     id: 2,
     value: "Medium",
     minDiversity: 4,
-    minLength: 8
+    minLength: 10,
   },
   {
     id: 3,
     value: "Strong",
     minDiversity: 4,
-    minLength: 10
-  }
+    minLength: 12,
+  },
 ];
 
-const passwordStrength = (password, options = defaultOptions, allowedSymbols = "!\"#\$%&'\(\)\*\+,-\./:;<=>\?@\[\\\\\\]\^_`\{|\}~") => {
+const owaspSymbols = "!\"#$%&'()*+,-./:;<=>?@[\\\\\\]^_`{|}~";
 
-  let passwordCopy = password || '';
+const passwordStrength = (
+  password,
+  options = defaultOptions,
+  restrictSymbolsTo
+) => {
+  options[0].minDiversity = 0;
+  options[0].minLength = 0;
 
-  options[0].minDiversity = 0,
-    options[0].minLength = 0;
+  // prevent [a-z] to match null and compute length
+  const _password = password ?? "";
 
   const rules = [
     {
+      key: "lowercase",
       regex: "[a-z]",
-      message: 'lowercase'
     },
     {
-      regex: '[A-Z]',
-      message: 'uppercase'
+      key: "uppercase",
+      regex: "[A-Z]",
+    },
+    {
+      key: "number",
+      regex: "[0-9]",
     },
     {
-      regex: '[0-9]',
-      message: 'number'
+      key: "symbol",
+      regex: restrictSymbolsTo ? `[${restrictSymbolsTo}]` : "[^a-zA-Z0-9]",
```

Comment:
it's not documented that `restrictSymbolsTo` needs to be a valid string for inside a character class in a regular expression. i'd recommend using [`escape-string-regexp`](https://www.npmjs.com/package/escape-string-regexp) here (then cleaning up the string above to get rid of the extra `\`s

Reply:
for example, passing `$][&` for `restrictSymbolsTo` would require that the password match `[$][&]`, so the strings `x$x`, `x&x`, and `x&$x` would not match, but `x$&x` would.

Reply:
additionally, `,-.` in the `owaspSymbols` string will _technically_ get interpreted as "any character from `,` to `.`" because of the `-`. in reality, the only character between `,` and `.` in the ascii table is `-`, so it's fine, but this feels like a potential bug if the symbols in that string ever get rearranged. (but escaping the string like i laid out above would solve this problem)

Reply:
You are right. I added the library which is extremely tiny. I added the necessary information in the doc.